Tests/tests/DataStore2.spec.lua

```diff
@@ -79,6 +79,53 @@ return function()
 				expect(DataStore2(nonNilKey, fakePlayer):Get("badDefault")).to.equal("abc")
 			end)
 
+			it("should validate Set", function()
+				local dataStore = DataStore2(UUID(), fakePlayer)
+
+				local function testValidator(dataToValidate)
+					if dataToValidate == "yepp" then
+						return true
+					elseif dataToValidate == "definitelyNot" then
+						return false, "A validation error message"
+					end
+
+					return false
+				end
+
+				dataStore:SetValidator(testValidator)
+				expect(dataStore:Set("nope")).to.throw("Attempted to set data store to an invalid value during :Set")

```

Comment:
This isn't how `.to.throw` works, if `dataStore:Set("nope")` errors then it would error the entire script!

![image](https://user-images.githubusercontent.com/3190756/104105594-16f08200-5264-11eb-9f02-4a36527493c7.png)

Reply:
You need the `function` there because that's what's actually pcalled. The way you have it now looks like `:Set` returns a function that you're checking.